pkg/imgpkg/registry/registry.go

```diff
@@ -203,6 +209,14 @@ func NewSimpleRegistryWithTransport(opts Opts, rTripper http.RoundTripper, regOp
 // the image provided
 // A Registry need to be provided as the first parameter or the function will panic
 func (r SimpleRegistry) CloneWithSingleAuth(imageRef regname.Tag) (Registry, error) {
+	if r.keychain == nil { // If no keychain is present it assumes NewBasicRegistry was used to create the Registry. So we short circuit this execution
```

Comment:
From a domain perspective, what does it mean for `r.keychain` to be `nil`?

I believe it means that either no keychain is required _or_ the keychain will be created using the `...regremote.Option`s.

As long as those two are the only scenarios where `r.keychain` can be `nil`, this checks out.

Are these the only scenarios where `r.keychain` can be `nil`?

Reply:
The only situation where this can effectively happen is when you create the `SimpleRegistry` using `NewBasicRegistry` because the other constructors always create a keychain